DSpace/dspace-angular
# Communities and collections not shown right after creation

## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #842 

## Description

After a collection or community is created or deleted, href request to its parent is removed from requests cache, so that list of communities or collections belonging to a given community, at any level, from top level on, has to be reloaded and new collection / community is immediately displayed in case of creation or not displayed anymore in case of deletion.

## Instructions for Reviewers

a new function named `refreshCache` has been added to `CreateComColPageComponent` and `DeleteComColPageComponent`, and it is called by the `onSubmit` function of `CreateComColPageComponent` during the flow that observes result of `DSpaceObject` creation performed by `dsoDataService`, and by the `onConfirm` function of `DeleteComColPageComponent` during the flow that observes the result of `DSpaceObject` deletion performed by `dsoDataService`.


List of changes in this PR:
* `refreshCache` function in `CreateComColPageComponent` and its invocation
* `refreshCache` function in `DeleteComColPageComponent` and its invocation

To reproduce the bug: 

* Create a subcommunity within a community.
* Click on "All of DSpace" on the top of the page.
* Select the "parent-community" within which you added the subcommunity.
* The just created subcommunity doesn't appear in the list of the subcommunities.
* Delete a subcommunity within a community.
* Click on "All of DSpace" on the top of the page.
* Select the "parent-community" within which you deleted the subcommunity. 
* The just deleted subcommunity still appears in the list of subcommunities
* Create a collection within a community.
* Click on "All of DSpace" on the top of the page.
* Select the "parent-community" within which you added the collection.
* The just created collection doesn't appear in the list of the community collections.
* Delete a collection within a community.
* Click on "All of DSpace" on the top of the page.
* Select the "parent-community" within which you deleted the collection. 
* The just deleted collection still appears in the list of collections

To test changes: 

* Create a new top level community
* Newly created top level community should appear in dspace home page and in community-list page list
* Delete created top level community
* Deleted community should not appear anymore in DSpace home page and in community-list page list
* Create a new subcommunity
* Newly created subcommunity should appear immediately as child of its parent community in community-list page and between 'Communities of this community' section in community page
* Delete newly created subcommunity
* Deleted sub-community should not appear anymore  as child of its parent community in community-list page and between 'Communities of this community' section in community page
* Create a new collection of a Community
* Newly created collection should immediately appear between 'Collections of this community' section in community page
* Delete newly created collection
* Deleted collection should not appear anymore between 'Collections of this community' section in community page.



## Checklist
